bftengine/src/bftengine/messages/ReplicasRestartReadyProofMsg.cpp

```diff
@@ -0,0 +1,125 @@
+                                                           SeqNum seqNum,
+                                                           const concordUtils::SpanContext& spanContext)
+    : MessageBase(senderId,
+                  MsgCode::ReplicasRestartReadyProof,
+                  spanContext.data().size(),
+                  ReplicaConfig::instance().getmaxExternalMessageSize() - spanContext.data().size()) {
+  b()->genReplicaId = senderId;
+  b()->seqNum = seqNum;
+  b()->elementsCount = 0;
+  b()->locationAfterLast = 0;
+  std::memcpy(body() + sizeof(Header), spanContext.data().data(), spanContext.data().size());
+}
+const uint32_t ReplicasRestartReadyProofMsg::getBodySize() const {
+  uint32_t bodySize = b()->locationAfterLast;
+  if (bodySize == 0) bodySize = sizeof(Header) + spanContextSize();
+  return bodySize;
+}
+
+ReplicasRestartReadyProofMsg* ReplicasRestartReadyProofMsg::create(ReplicaId id,
+                                                                   SeqNum s,
+                                                                   const concordUtils::SpanContext& spanContext) {
+  ReplicasRestartReadyProofMsg* m = new ReplicasRestartReadyProofMsg(id, s, spanContext);
+  return m;
+}
+
+void ReplicasRestartReadyProofMsg::addElement(std::unique_ptr<ReplicaRestartReadyMsg>& restartMsg) {
+  if (b()->locationAfterLast == 0)  // if this is the first element
+  {
+    ConcordAssert(b()->elementsCount == 0);
+    b()->locationAfterLast = sizeof(Header) + spanContextSize();
+  }
+  uint32_t requiredSpace = b()->locationAfterLast + restartMsg->size();
+  ConcordAssertLE((size_t)(requiredSpace + SigManager::instance()->getMySigLength()), (size_t)internalStorageSize());
+  std::memcpy(body() + b()->locationAfterLast, restartMsg->body(), restartMsg->size());
+  b()->elementsCount += 1;
+  b()->locationAfterLast = requiredSpace;
+}
+// +---------------------------------------------------------------+--------------+
+// | Msg header(genReplicaId, seqNum, eleCount, locationAfterLast) | Span Context |
+// +---------------------------------------------------------------+--------------+
+// |  Element1(replicaId, seqNum, sigLen, sigBody)  |
+// +------------------------------------------------+
+// |     .................................          |
+// +------------------------------------------------+
+// |  Elementn(replicaId, seqNum, sigLen, sigBody)  |
+// +------------------------------------------------+
+
+void ReplicasRestartReadyProofMsg::finalizeMessage() {
+  auto bodySize = getBodySize();
+  setMsgSize(bodySize);
+  shrinkToFit();
+}
+
+void ReplicasRestartReadyProofMsg::validate(const ReplicasInfo& repInfo) const {
+  auto sigManager = SigManager::instance();
+  if (size() < sizeof(Header) + spanContextSize() || !repInfo.isIdOfReplica(idOfGeneratedReplica()) ||
+      idOfGeneratedReplica() == repInfo.myId())
+    throw std::runtime_error(__PRETTY_FUNCTION__ + std::string(": basic validations"));
```

Comment:
same comment as before. for all exceptions in this function: I assume you are cathing this and dropping/ignoring the message - right?

Reply:
yes